README.md

```diff
@@ -1,2 +1,424 @@
+      initialArray.push(2);
+      expect(initialArray).toEqual([1, 2]);
+    });
+  });
+});
+```
+
+#### Better:
+
+```javascript
+describe('Array.prototype', function() {
+  describe('.push(x)', function() {
+    var initialArray;
+
+    beforeAll(function() {
+      initialArray = [1]; // Arrange
+
+      initialArray.push(2); // Act
+    });
+
+    it('appends x to the end of the Array', function() {
+      expect(initialArray).toEqual([1, 2]); // Assert
+    });
+  });
+});
+```
+
+### Don't Repeat Yourself
+
+Use `before`/`after` blocks to DRY up repeated setup, teardown, and action code.
+
+#### Why?
+
+* Keeps test suite more concise and readable
+* Changes only need to be made in one place
+* Unit tests are not exempt from coding best practices
+
+#### Bad:
+
+```javascript
+describe('Array.prototype', function() {
+  describe('.push(x)', function() {
+    it('appends x to the end of the Array', function() {
+      var initialArray = [1];
+      initialArray.push(2);
+      expect(initialArray).toEqual([1, 2]);
+    });
+
+    it('returns x', function() {
+      var initialArray = [1];
+      expect(initialArray.push(2)).toBe(2);
+    });
+  });
+});
+```
+
+#### Better:
+
+```javascript
+describe('Array.prototype', function() {
+  describe('.push(x)', function() {
+    var initialArray,
+        pushResult;
+
+    beforeAll(function() {
+      initialArray = [1];
+
+      pushResult = initialArray.push(2);
+    });
+
+    it('appends x to the end of the Array', function() {
+      expect(initialArray).toEqual([1, 2]);
+    });
+
+    it('returns x', function() {
+      expect(pushResult).toBe(2);
+    });
+  });
+});
+```
+
+### `this` Is How We Do It
+
+Use `this` to share variables between `it` and `before`/`after` blocks.
+
+#### Why?
+
+* Declare and initialize variables on one line
+* Jasmine automatically cleans the `this` object between specs to avoid state leak
+
+#### Bad:
+
+```javascript
+describe('Array.prototype', function() {
+  describe('.push(x)', function() {
+    var initialArray,
+        pushResult;
+
+    beforeAll(function() {
+      initialArray = [1];
+
+      pushResult = initialArray.push(2);
+    });
+
+    it('appends x to the end of the Array', function() {
+      expect(initialArray).toEqual([1, 2]);
+    });
+
+    it('returns x', function() {
+      expect(pushResult).toBe(2);
+    });
+  });
+});
+```
+
+#### Better:
+
+```javascript
+describe('Array.prototype', function() {
+  describe('.push(x)', function() {
+    beforeAll(function() {
+      this.initialArray = [1];
+
+      this.pushResult = this.initialArray.push(2);
+    });
+
+    it('appends x to the end of the Array', function() {
+      expect(this.initialArray).toEqual([1, 2]);
+    });
+
+    it('returns x', function() {
+      expect(this.pushResult).toBe(2);
+    });
+  });
+});
+```
+
+### Avoid the `All`s
+
+Prefer `beforeEach/afterEach` blocks over `beforeAll/afterAll` ones. The latter are not reset between tests.
```

Comment:
The idea of using `beforeAll` in the other examples was to have a theme of "fixing as we go" but I can see it being confusing. I guess `beforeAll`/`afterAll` do have there use cases (otherwise they wouldn't exist). Perhaps this rule should be removed all together.